state/factory/workingset.go

```diff
@@ -88,27 +89,54 @@ func newWorkingSet(
 	root hash.Hash256,
 	saveHistory bool,
 ) (WorkingSet, error) {
-	ws := &workingSet{
-		finalized:   false,
-		blockHeight: height,
-		saveHistory: saveHistory,
-		trieRoots:   make(map[int]hash.Hash256),
-		cb:          db.NewCachedBatch(),
-		dao:         kv,
+	opts := []db.KVStoreBatchOption{db.DigestFilterOption(func(wi *db.WriteInfo) bool {
+		switch wi.Namespace() {
+		case evm.PreimageKVNameSpace, AccountTrieNamespace:
```

Comment:
as we discussed, PreimageKVNameSpace should not be excluded

so it's just 
return wi.Namespace() == AccountTrieNamespace